device/cuda/src/utils/barrier.hpp

```diff
@@ -1,29 +1,28 @@
 /** TRACCC library, part of the ACTS project (R&D line)
  *
- * (c) 2023 CERN for the benefit of the ACTS project
+ * (c) 2023-2024 CERN for the benefit of the ACTS project
  *
  * Mozilla Public License Version 2.0
  */
 
 #pragma once
 
-// Project include(s).
-#include "traccc/definitions/qualifiers.hpp"
-
 namespace traccc::cuda {
 
 struct barrier {
-    TRACCC_DEVICE
-    void blockBarrier() { __syncthreads(); }
+    __device__ inline void blockBarrier() const { __syncthreads(); }
 
-    TRACCC_DEVICE
-    bool blockAnd(bool predicate) { return __syncthreads_and(predicate); }
+    __device__ inline bool blockAnd(bool predicate) const {
+        return __syncthreads_and(predicate);
+    }
 
-    TRACCC_DEVICE
-    bool blockOr(bool predicate) { return __syncthreads_or(predicate); }
+    __device__ inline bool blockOr(bool predicate) const {
+        return __syncthreads_or(predicate);
+    }
 
-    TRACCC_DEVICE
-    int blockCount(bool predicate) { return __syncthreads_count(predicate); }
+    __device__ inline int blockCount(bool predicate) const {
+        return __syncthreads_count(predicate);
+    }
```

Comment:
Hmm, why? The previous version felt more consistent with the rest of the code.

Reply:
Why should we use a macro in a file that can only be used in CUDA device code? Much like how I removed `TRACCC_DEVICE` from the SYCL code (as it served really no purpose there), I thought this would also be a good simplification.

The macros are really only needed in the shared code. CUDA specific code should be free to use `__device__`, `__global__`, etc.